### PR TITLE
Change folder structure

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Changed
 - Changed internal folder structure to `src`, `tests` and `docs`
```

## Subject

Adopt the new folder structure discussed on the [dev-kit](https://github.com/sonata-project/dev-kit/issues/278)

### CODE REVIEW

**Comment by greg0ire** [comment]
> I am targeting this branch, because this is a BC break.

Is it really though? Autoloading is taken care of...

**Comment by core23** [comment]
You're right, we've already done this on a stable branch (https://github.com/sonata-project/cache/pull/43)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ Just like php dependencies can be managed with Composer, python dependencies can
 To get sphinx, simply run the following command.
 
 ```bash
-pip install --requirement Resources/doc/requirements.txt --user
+pip install --requirement docs/requirements.txt --user
```
Comment (greg0ire) [inline CONTRIBUTING.md]: This file is managed through dev-kit. Maybe we should recommend `make docs` instead?

Comment (core23) [reply CONTRIBUTING.md]: We already recommend `make docs`, if you read the whole paragraph.

Comment (greg0ire) [reply CONTRIBUTING.md]: Oh yeah I'm tired, that's not about producing the docs, it's about installing sphinx. Sorry for the trouble.

**Comment by OskarStark** [comment]
Great job @core23 👍

**Pull request merged**